lib/iris/fileformats/netcdf.py

```diff
@@ -1299,6 +1299,14 @@ def _cf_coord_identity(self, coord):
             elif coord.standard_name == "longitude":
                 units = 'degrees_east'
 
+        elif isinstance(coord.coord_system, iris.coord_systems.RotatedGeogCS):
+            units = 'degrees'
+
+        elif isinstance(coord.coord_system,
+                        (iris.coord_systems.TransverseMercator,
+                         iris.coord_systems.LambertAzimuthalEqualArea)):
+            units = 'm'
+
```

Comment:
In fact, if you look at Example 5.6, that's a rotated_lat_lon grid with units of "degrees". Also Example 5.10 is a tranverse_mercator grid with units of "m".